lib/internal/Magento/Framework/App/DeploymentConfig/Reader.php

```diff
@@ -81,47 +81,57 @@ public function getFiles()
-     * @throws RuntimeException If file is invalid
-     * @throws \Exception If file key is not correct
-     * @see FileReader
+     * @param $fileKey
+     * @return array|mixed
+     * @throws \Magento\Framework\Exception\FileSystemException
+     * @throws \Magento\Framework\Exception\RuntimeException
      */
     public function load($fileKey = null)
     {
+        static $cache = [];
+
         $path = $this->dirList->getPath(DirectoryList::CONFIG);
         $fileDriver = $this->driverPool->getDriver(DriverPool::FILE);
         $result = [];
+
         if ($fileKey) {
             $filePath = $path . '/' . $this->configFilePool->getPath($fileKey);
-            if ($fileDriver->isExists($filePath)) {
-                $result = include $filePath;
-                if (!is_array($result)) {
-                    throw new RuntimeException(new Phrase("Invalid configuration file: '%1'", [$filePath]));
-                }
-            }
+            $cache = $this->getFileCache($cache, $filePath, $fileDriver);
+            $result = $cache[$filePath] ?: [];
         } else {
             $configFiles = $this->getFiles();
             foreach ($configFiles as $file) {
                 $configFile = $path . '/' . $file;
-                if ($fileDriver->isExists($configFile)) {
-                    $fileData = include $configFile;
-                    if (!is_array($fileData)) {
-                        throw new RuntimeException(new Phrase("Invalid configuration file: '%1'", [$configFile]));
-                    }
-                } else {
-                    continue;
-                }
-                if ($fileData) {
-                    $result = array_replace_recursive($result, $fileData);
+                $cache = $this->getFileCache($cache, $configFile, $fileDriver);
+                if ($cache[$configFile]) {
+                    $result = array_replace_recursive($result, $cache[$configFile]);
                 }
             }
         }
         return $result ?: [];
     }
+
+    /**
+     * @param array $cache
+     * @param string $filePath
+     * @param \Magento\Framework\Filesystem\DriverInterface $fileDriver
+     * @return array
+     * @throws \Magento\Framework\Exception\FileSystemException
+     * @throws \Magento\Framework\Exception\RuntimeException
+     */
+    public function getFileCache(array $cache, string $filePath, \Magento\Framework\Filesystem\DriverInterface $fileDriver): array
+    {
+        if (!isset($cache[$filePath])) {
+            if ($fileDriver->isExists($filePath)) {
+                $fileData = include $filePath;
+                if (!is_array($fileData)) {
+                    throw new RuntimeException(new Phrase("Invalid configuration file: '%1'", [$filePath]));
+                }
+                $cache[$filePath] = $fileData;
+            } else {
+                $cache[$filePath] = null;
+            }
+        }
+        return $cache;
+    }
 }
```

Comment:
It would be beneficial if all namespaces were imported:
\Magento\Framework\Filesystem\DriverInterface
\Magento\Framework\Exception\FileSystemException
\Magento\Framework\Exception\RuntimeException

Reply:
What would be the benefit, though?

Reply:
@bgorski It is not related to performance, but rather improving readability and consistency in the code. As you can see, the codebase was recently refactored to import namespaces. Specifically, all namespaces in this class were imported.

Reply:
Additionally, it also avoids the line length rule since the following line exceeds 120 characters:
```
    public function getFileCache(array $cache, string $filePath, \Magento\Framework\Filesystem\DriverInterface $fileDriver): array
```